src/pudl_archiver/depositors/zenodo.py

```diff
@@ -224,91 +224,42 @@ async def get_new_version(
-            else:
-                return deposition
-
-        url = f"{self.api_root}/deposit/depositions/{deposition.id_}/actions/newversion"
-
-        # create a new unpublished deposition version
-        try:
-            response = await self.request(
-                "POST",
-                url,
-                log_label="Creating new version",
-                headers=self.auth_write,
-                retry_count=2,
-            )
-        # Except if abandoned in progress draft
-        except ZenodoClientError as excinfo:
-            if (
-                clobber
-                and "remove all files first" in excinfo.errors[0]["messages"][0].lower()
-            ):
-                logger.info("Delete abandoned version and create new one.")
-                existing_deposition = await self.get_deposition(
-                    str(deposition.conceptdoi)
-                )
-                new_draft = Record(
-                    **await self.request(
-                        "POST",
-                        f"{self.api_root}/records/{existing_deposition.id_}/versions",
-                        log_label=f"Get existing draft deposition for {existing_deposition.id_}",
-                        headers=self.auth_write,
-                    )
-                )
-                await self.request(
-                    "DELETE",
-                    new_draft.links.self,
-                    log_label=f"Delete existing draft deposition for {existing_deposition.id_}",
-                    headers=self.auth_write,
-                    parse_json=False,
-                )
-            response = await self.request(
+        # just get the new draft from new API.
+        new_draft_record = Record(
+            **await self.request(
                 "POST",
-                url,
-                log_label="Creating new version",
+                f"{self.api_root}/records/{deposition.id_}/versions",
+                log_label=f"Get existing draft deposition for {deposition.id_}",
                 headers=self.auth_write,
             )
-
-        old_metadata = deposition.metadata.dict(by_alias=True)
-        new_version = Deposition(**response)
-
-        source_metadata = new_version.metadata.dict(by_alias=True)
-
-        # If version not in response for new version, get from most recent deposition
-        if source_metadata["version"] is None:
-            source_metadata["version"] = old_metadata["version"]
-
-        metadata = {}
-        for key, val in source_metadata.items():
-            if key not in ["doi", "prereserve_doi", "publication_date"]:
-                metadata[key] = val
-
-        previous = semantic_version.Version(source_metadata["version"])
-        version_info = previous.next_major()
-
-        metadata["version"] = str(version_info)
+        )
+        # then, get the deposition based on that ID.
+        new_draft_deposition = await self.get_deposition_by_id(new_draft_record.id_)
```

Comment:
clarifying question.... i get confused about all the different kinds of ids flying around. If the `new_draft_record.id` going to try to make a draft archive based on the same underlying ... idk what it is called ... "core" id (or deposition id?)... the one that links CEMS version 1 to CEMS version 2?

Reply:
Ah, so what's happening here is:

1. We're using the New API for creating a new draft, since it's less flaky - which returns a slightly different data shape than the Old API.
2. But, everywhere else, we're using the Old API. So we need to use the ID that we just got back from the New API endpoint, and feed it through the Old API to get the right shape back.

Reply:
wait the new and the old api are both running and interacting with the same draft records?!? do we expect the new api is actually stable enough to rely on? (good news that it is less flaky though!)

Reply:
So, there are two ways to create a new draft: the old API and the new API.

The main relevant difference here is that the old API will *fail* if there's already a draft lying around, and the new API will just return that draft.

We *could* use the old API and catch the exception, except... then we have to go find the existing draft through some other means. We used to do this by GETting the "list depositions with an optional query string" endpoint in the legacy API. We would ask it for "give us the latest deposition that belongs to this concept DOI" and it would give us the draft back.

But, now that endpoint doesn't work. So the ONLY WAY we can get the latest draft in that case is actually by using the new API.

So our options are:

1. Use the new API and just handle things breaking when the behavior changes
2. Use the old API, but then rely on the new API anyways when the old API fails...

So we're sort of stuck using the new API either way.

Reply:
Yeah I don't love mixing API's, but overall I think it seems simpler than doing the error handling ourselves, so overall I'm ok with this if it has been working consistently.